examples/plot_compare_reduction.py

```diff
@@ -72,4 +86,42 @@
 plt.ylabel('Digit classification accuracy')
 plt.ylim((0, 1))
 plt.legend(loc='upper left')
+
+###############################################################################
+# Illustration of ``CachedPipeline`` instead of ``Pipeline``
+###############################################################################
+# It is sometimes interesting to store the state of a specific transformer
+# since it could be used again. Using a pipeline in ``GridSearchCV`` triggers
+# such situations. Therefore, we replace ``Pipeline`` with ``CachedPipeline``
+# to memoize the transfomers within the pipeline.
+#
+# .. warning::
+#     Note that this example is, however, only an illustration since for this
+#     specific case fitting PCA is not necessarily slower than loading the
+#     cache. Hence, use ``CachedPipeline`` when the fitting of a transformer
+#     is costly.
+
+from tempfile import mkdtemp
+from sklearn.pipeline import CachedPipeline
+from sklearn.externals.joblib import Memory
+
+# Create a temporary folder to store the transformers of the pipeline
+cachedir = mkdtemp()
```

Comment:
`cachedir` should be deleted (for instance with `shutil.rmtree`) at the end of the script.